spec/support/helpers.rb

```diff
@@ -216,10 +214,12 @@ def rake
       "#{Gem.ruby} -S #{ENV["GEM_PATH"]}/bin/rake"
     end
 
-    def sys_exec(cmd)
+    def sys_exec(cmd, env = {})
```

Comment:
Do you think the added readability is worth losing the consistency?

None of the helper methods in this file use keyword args at the moment, so if I quickly skim through the file I might assume the `sys_exec` method is getting an optional hash like every other method and get it wrong.

Reply:
We haven't been able to use keyword arguments before because of having to maintain compatibility with Ruby 1.8, 1.9.

I think it will greatly help with readability and reduce helper complexity

Example:

    bundle! "install", :standalone => true 

Unless you're pretty familiar with bundle install and its options, it's not super clear what the purpose of the `standalone` option is. With keyword args, we make it more explicit to the reader:

    bundle! "install", args: { standalone: true }

In terms of the complexity of the helper methods, take the `bundle` helper for example:

(haven't checked if the code actually works, but i trust you get the point)

Before:
```ruby

    def bundle(cmd, options = {})
      with_sudo = options.delete(:sudo)
      sudo = with_sudo == :preserve_env ? "sudo -E" : "sudo" if with_sudo

      bundle_bin = options.delete("bundle_bin") || bindir.join("bundle")

      if system_bundler = options.delete(:system_bundler)
        bundle_bin = system_bundle_bin_path
      end

      env = options.delete(:env) || {}
      env["PATH"].gsub!("#{Path.root}/exe", "") if env["PATH"] && system_bundler

      requires = options.delete(:requires) || []
      requires << "support/hax"

      artifice = options.delete(:artifice) do
        if RSpec.current_example.metadata[:realworld]
          "vcr"
        else
          "fail"
        end
      end
      if artifice
        requires << File.expand_path("../artifice/#{artifice}", __FILE__)
      end

      requires_str = requires.map {|r| "-r#{r}" }.join(" ")

      load_path = []
      load_path << lib unless system_bundler
      load_path << spec
      load_path_str = "-I#{load_path.join(File::PATH_SEPARATOR)}"

      env = env.map {|k, v| "#{k}='#{v}'" }.join(" ")

      args = options.map do |k, v|
        case v
        when nil
          next
        when true
          " --#{k}"
        when false
          " --no-#{k}"
        else
          " --#{k} #{v}"
        end
      end.join

      cmd = "#{env} #{sudo} #{Gem.ruby} #{load_path_str} #{requires_str} #{bundle_bin} #{cmd}#{args}"
      sys_exec(cmd) {|i, o, thr| yield i, o, thr if block_given? }
    end
```

After:
```ruby

    def bundle(cmd, sudo: false, env: {}, args: {}, requires: [], artifice: nil, bundle_bin: nil, system_bundler: nil)
      with_sudo = sudo == :preserve_env ? "sudo -E" : "sudo" if sudo
      bundle_bin ||= bindir.join("bundle")
      system_bundler ||= system_bundle_bin_path
      artifice ||= RSpec.current_example.metadata[:realworld] ? "vcr" : "fail"

      env["PATH"].gsub!("#{Path.root}/exe", "") if env["PATH"] && system_bundler

      requires << "support/hax"
      requires << File.expand_path("../artifice/#{artifice}", __FILE__) if artifice 

      requires_str = requires.map {|r| "-r#{r}" }.join(" ")

      load_path = []
      load_path << lib unless system_bundler
      load_path << spec
      load_path_str = "-I#{load_path.join(File::PATH_SEPARATOR)}"

      bundle_env = env.map {|k, v| "#{k}='#{v}'" }.join(" ")

      bundle_args = args.map do |k, v|
        case v
        when nil
          next
        when true
          " --#{k}"
        when false
          " --no-#{k}"
        else
          " --#{k} #{v}"
        end
      end.join

      cmd = "#{with_sudo} #{Gem.ruby} #{load_path_str} #{requires_str} #{bundle_bin} #{cmd}#{bundle_args}"
      sys_exec(cmd, env: bundle_env) {|i, o, thr| yield i, o, thr if block_given? }
    end
```

I understand the consistency part but i think the effort is worth the trade-off in this particular case.

Reply:
Oh, yes, absolutely! I failed to explain myself, I totally agree with keyword args making things better.

My point was that the burden on making this migration shouldn't fail on @janpio, but on us, and when we do it, we should migrate all helpers at once, not just this single one.

Reply:
That makes sense. I'm happy for another PR to tackle this issue.